commit

yanakrupets opened the pull request.

yanakrupets committed:
commit

VladickSharpist commented on Assets/Scripts/Editor/ColliderSaverTool.cs:
```diff
@@ -0,0 +1,126 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace Editor
```
namespace Editor;

VladickSharpist commented on Assets/Scripts/Editor/ColliderSaverTool.cs:
```diff
@@ -0,0 +1,126 @@
+                .ObjectField("Sprite Renderer", _overlapSpriteRenderer, typeof(SpriteRenderer), true);
+
+            if (GUILayout.Button("Save Collider Paths"))
+            {
```
на однострочный выхов скобки не пишем

VladickSharpist commented on Assets/Scripts/Editor/ColliderSaverTool.cs:
```diff
@@ -0,0 +1,126 @@
+            }
+        
+            if (GUILayout.Button("Update Collider"))
+            {
```
тоже самое скобка

VladickSharpist commented on Assets/Scripts/Editor/ColliderSaverTool.cs:
```diff
@@ -0,0 +1,126 @@
+            _overlapSpriteRenderer = (SpriteRenderer)EditorGUILayout
+                .ObjectField("Sprite Renderer", _overlapSpriteRenderer, typeof(SpriteRenderer), true);
+
+            if (GUILayout.Button("Save Collider Paths"))
```
заменить на свитч

VladickSharpist commented on Assets/Scripts/LevelGenerator.cs:
```diff
@@ -0,0 +1,108 @@
+    private void GenerateObjects<TObject>(
+        LandscapeLayer landscapeLayer, 
+        TObject prefab,
+        //action -> delegate
+        Action<TObject, List<ColliderPathPoints>, Sprite, Vector2, string> initialize)
```
тип делеагата

VladickSharpist commented on Assets/Scripts/ScriptableObjects/ColliderData.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+        [field: SerializeField] public List<ColliderPath> ColliderPaths { get; private set; }
+
+        public ColliderPath this[OverlapType overlapType]
```
почему не ColliderPath? если FirstOrDefaut может вернуть налл

VladickSharpist replied:
Почему только гет, если только гет то вот так public ColliderPath this[OverlapType overlapType] => ....

VladickSharpist commented on Assets/Scripts/ScriptableObjects/ColliderData.cs:
```diff
@@ -0,0 +1,45 @@
+    [CreateAssetMenu(fileName = "Collider Data", menuName = "Configs/Collider Data")]
+    public class ColliderData : ScriptableObject
+    {
+        [field: SerializeField] public List<ColliderPath> ColliderPaths { get; private set; }
```
private set можно не писать

VladickSharpist commented on Assets/Scripts/ScriptableObjects/ColliderData.cs:
```diff
@@ -0,0 +1,45 @@
+
+        public ColliderPath this[OverlapType overlapType]
+        {
+            get { return ColliderPaths.FirstOrDefault(path => path.OverlapType == overlapType); }
```
lamda get => ColliderPaths...

VladickSharpist commented on Assets/Scripts/ScriptableObjects/ColliderData.cs:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    [Serializable]
+    public class ColliderPathPoints
```
если паблик то отдельный файл

VladickSharpist commented on Assets/Scripts/ScriptableObjects/ColliderData.cs:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    [Serializable]
+    public class ColliderPath
```
если паблик то отдельный файл

VladickSharpist commented on Assets/Scripts/LevelGenerator.cs:
```diff
@@ -0,0 +1,108 @@
+        Action<TObject, List<ColliderPathPoints>, Sprite, Vector2, string> initialize)
+        where TObject : MonoBehaviour
+    {
+        var data = placementData.FirstOrDefault(data => data.landscapeLayer == landscapeLayer);
```
а может ли тут быть два элемента с одним landscapeLayer, если да то точно ли правильно написана логика, если нет то почему не SingleOrDefault

VladickSharpist commented on Assets/Scripts/ScriptableObjects/GraphicData.cs:
```diff
@@ -0,0 +1,44 @@
+        [SerializeField] private List<Sprite> backgroundSprites;
+        [SerializeField] private List<Sprite> landscapeBackgroundSprites;
+
+        public List<OverlapGraphic> GetOverlapGraphic(LandscapeLayer landscapeLayer)
+        {
+            return overlapGraphics
+                .Where(overlap => (overlap.LandscapeLayer & (LandscapeLayerFlags)landscapeLayer) != 0)
+                .ToList();
+        }
```
зачем тебе тут лист, судя по использованиям ты его не меняешь, верни массив, обёрнутый в IReadOnlyCollection\<OverlapGraphics\>

VladickSharpist replied:
\+ используй =>

VladickSharpist commented on Assets/Scripts/ScriptableObjects/GraphicData.cs:
```diff
@@ -0,0 +1,44 @@
+            return backgroundSprites[Random.Range(0, backgroundSprites.Count)];
+        }
+        
+        public Sprite GetRandomLandscapeBackgroundSprite()
+        {
+            return landscapeBackgroundSprites[Random.Range(0, backgroundSprites.Count)];
+        }
```
=>

VladickSharpist commented on Assets/Scripts/ScriptableObjects/GraphicData.cs:
```diff
@@ -0,0 +1,44 @@
+                .ToList();
+        }
+
+        public Sprite GetRandomBackgroundSprite()
+        {
+            return backgroundSprites[Random.Range(0, backgroundSprites.Count)];
+        }
```
=>

VladickSharpist commented on Assets/Scripts/ScriptableObjects/PositionData.cs:
```diff
@@ -0,0 +1,27 @@
+    {
+        [field: SerializeField] public List<OverlapPosition> OverlapPositions { get; private set; }
+
+        public OverlapPosition this[OverlapType overlapType]
+        {
+            get { return OverlapPositions.FirstOrDefault(overlap => overlap.OverlapType == overlapType); }
+        }
```
=>

VladickSharpist replied:
и точно ли first or default

VladickSharpist commented on Assets/Scripts/ScriptableObjects/PositionData.cs:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    [Serializable]
+    public class OverlapPosition
+    {
+        [field: SerializeField] public OverlapType OverlapType { get; private set; }
+        [field: SerializeField] public float Y { get; private set; }
+    }
+}
```
мб private или отдельный файл?

yanakrupets committed:
some fix

yanakrupets committed:
pool & fabric

+ test class and objects on scene

VladickSharpist commented on Assets/Scripts/LandscapeItem.cs:
```diff
@@ -0,0 +1,33 @@
+    
+    // serialized particle system
+    
+    public void Initialize(List<ColliderPathPoints> paths, 
```
перенос

VladickSharpist replied:
IReadonlyCollection вместо лист

VladickSharpist replied:
или вообще массив, но смотри что бы он не скопировался

VladickSharpist commented on Assets/Scripts/ColliderPath.cs:
```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
+
+[Serializable]
+public class ColliderPath
```
мб структура?

VladickSharpist commented on Assets/Scripts/LandscapeItem.cs:
```diff
@@ -1,27 +1,34 @@
-    [SerializeField] private SpriteRenderer _spriteRenderer;
-    [SerializeField] private PolygonCollider2D _collider;
+    private SpriteRenderer _spriteRenderer;
+    private PolygonCollider2D _polygonCollider;
```
_polygonCollider -> _polygon2dCollider

VladickSharpist commented on Assets/Scripts/Pool/ObjectPool.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace Pool
+{
+    public class ObjectPool<T> where T : MonoBehaviour, IPoolableItem
```
IPoolableItem очень плохое нахвание, обжект пул управляет пулабл айтемами ну по логике как мне кажется, а у тебя получается что обжект пул это и есть пулабл айтем, странно

VladickSharpist commented on Assets/Scripts/Pool/ObjectPool.cs:
```diff
@@ -0,0 +1,44 @@
+            else
+            {
+                var obj = _factory.Create();
+                obj.gameObject.SetActive(true);
+                obj.OnGet();
```
почему то мне кажется что он гет должен быть раньше чем объект появится как думаешь?

VladickSharpist commented on Assets/Scripts/Pool/ObjectPool.cs:
```diff
@@ -0,0 +1,44 @@
+            obj.OnReturn();
+            obj.gameObject.SetActive(false);
+            obj.transform.SetParent(_factory.Parent);
+            obj.transform.localPosition = Vector3.zero;
```
смотри, в целом я понимаю для чего это, НО у тебя получается неявно меняются свойства объекта никак не относящиеся к пулу объектов. Пока конечно идеи у меня нет куда засунуть эту строку, как будто в OnReturn не получится

VladickSharpist commented on Assets/Scripts/Pool/ObjectPool.cs:
```diff
@@ -0,0 +1,44 @@
+        {
+            obj.OnReturn();
+            obj.gameObject.SetActive(false);
+            obj.transform.SetParent(_factory.Parent);
```
в чём прикол парента? для чего это вообще, ещё и в интерфейсе присутствует

VladickSharpist commented on Assets/Scripts/Factories/PrefabFactory.cs:
```diff
@@ -0,0 +1,25 @@
+        public T Create()
+        {
+            var obj = Object.Instantiate(_prefab, Parent);
+            obj.gameObject.SetActive(false);
```
видишь у тебя и фактори отвечает за сет эктив, и обжект пул отвечает за сет эктив, какая то хуйня

VladickSharpist commented on Assets/Scripts/Managers/PoolManager.cs:
```diff
@@ -0,0 +1,36 @@
+        [SerializeField] private EnemyItem enemyPrefab;
+        [SerializeField] private Transform enemyParent;
+
+        private readonly Dictionary<Type, object> _pools = new();
```
почему тут обжект, если можно поставить интерфейс, иначе я сюда могу засунуть любое говно, и при этом сделав нужный тип тебе не придётся кастить это когда достанешь

VladickSharpist commented on Assets/Scripts/Managers/PoolManager.cs:
```diff
@@ -0,0 +1,36 @@
+
+        public ObjectPool<T> GetPool<T>() where T : MonoBehaviour, IPoolableItem
+        {
+            return (ObjectPool<T>)_pools[typeof(T)];
```
нахуя тебе интерфейс если ты его не возвращаешь, старайся не возвращать конкретные типы, в данном случае уж точно

VladickSharpist commented on Assets/Scripts/Tests/PoolTest.cs:
```diff
@@ -0,0 +1,55 @@
+            var citizen = pool.Get();
+            _citizenItems.Enqueue(citizen);
+        
+            citizen.transform.SetParent(citizenParent);
+            citizen.transform.localPosition = Vector3.zero;
```
получается ещё и при использовании тебе придётся менять позицию и парента, помимо того что это в других местах меняется(в пуле и в фактори)

VladickSharpist commented on Assets/Scripts/Tests/PoolTest.cs:
```diff
@@ -0,0 +1,55 @@
+        [Button("Return Citizen")]
+        private void ReturnCitizen()
+        {
+            var pool = poolManager.GetPool<CitizenItem>();
+            pool.Return(_citizenItems.Dequeue());
```
при гете ты меняешь свойства, а при ретурне не надо, что то не то

yanakrupets committed:
some fix

yanakrupets committed:
Added animation and target placement logic

AnimationController, ILandscapeItem, LevelGenerator, GameManager, AimWeightData, GraphicData, AimWeight, AnimationData, TargetItem, TargetPlace, TargetZone + refactoring, removed IPoolableItem

yanakrupets committed:
Fix landscape colliders

Removed unnecessary colliders from back and front landscape (LandscapeItem, LevelGenerator, TargetZone)

yanakrupets committed:
Cursor update

yanakrupets committed:
update

Folders, bootstrap, shootable items logic change

yanakrupets committed:
animation controller refactoring

yanakrupets committed:
Yo commit

Input system, game state machine, ui controller, removed test, some small fix

yanakrupets committed:
enemies variations

Added graphic controller and sets of enemies

yanakrupets committed:
save

yanakrupets committed:
sounds

yanakrupets committed:
more graphic

added to graphic config

yanakrupets committed:
small fix

yanakrupets committed:
some updates

Added pixel font + ui button graphic, fix bug in GameManager -> AimCycleRoutine

yanakrupets committed:
some fix

Canvas scaler, uni task, cursor controller
